src/data/queries/alertSingle.ts

```diff
@@ -1,53 +1,69 @@
+import { QueryFormatter } from 'next-drupal-query'
+import { ParagraphAlertSingle } from '@/types/drupal/paragraph'
 import {
-  QueryData,
-  QueryFormatter,
-  QueryParams,
-  QueryOpts,
-} from 'next-drupal-query'
-import { drupalClient } from '@/lib/drupal/drupalClient'
+  AlertSingle,
+  AlertBlock,
+  AlertNonReusable,
+} from '@/types/formatted/alert'
+import { formatParagraph } from '@/lib/drupal/paragraphs'
 import { queries } from '.'
-import { ParagraphAlertSingle } from '@/types/drupal/paragraph'
-import { AlertSingle } from '@/types/formatted/alertSingle'
-import { transformAlertData } from '@/lib/drupal/query'
-
-export const params: QueryParams<null> = () => {
-  return queries
-    .getParams()
-    .addInclude([
-      'field_alert_block_reference',
-      'field_alert_block_reference.field_alert_content',
-      'field_alert_non_reusable_ref',
-      'field_alert_non_reusable_ref.field_va_paragraphs',
-    ])
-}
 
-type DataOpts = QueryOpts<null>
-
-export const data: QueryData<DataOpts, AlertSingle[]> = async (
-  opts
-): Promise<AlertSingle[]> => {
-  const entities = await drupalClient.getResourceCollection<AlertSingle[]>(
-    'paragraph--alert_single',
-    {
-      params: params().getQueryObject(),
-    }
-  )
-
-  return entities
-}
+// TODO:
+// We don't need this here:
+// export const params: QueryParams<null> = () => {
+//   return queries
+//     .getParams()
+//     .addInclude([
+//       'field_alert_block_reference',
+//       'field_alert_block_reference.field_alert_content',
+//       'field_alert_non_reusable_ref',
+//       'field_alert_non_reusable_ref.field_va_paragraphs',
+//     ])
+// }
+//
+// ...but it seems like we might want to define
+// paragraph-specific `includes` within the paragraph's
+// query file (this file), and then somehow reference these in
+// the node query when we need to include fields of a specific
+// paragraph type.
+//
+// E.g.
+// export const include = [
+//   'field_alert_block_reference',
+//   'field_alert_block_reference.field_alert_content',
+//   'field_alert_non_reusable_ref',
+//   'field_alert_non_reusable_ref.field_va_paragraphs',
+// ]
+//
+// /* Some util file*/
+// export const getIncludedSubFields = (
+//   parentField: string,
+//   subFields: string[]
+// ) => [parentField, ...subFields.map((subField) => `${parentField}.${subField}`)]
+//
+// /* src/data/queries/resourcesSupport.ts */
+// import { include } from '@/data/queries/alertSingle'
+// export const params: QueryParams<null> = () => {
+//   return queries.getParams().addInclude([
+//     ...getIncludedSubFields('field_alert_single', include)
+//   ])
+// }
 
-export const formatter: QueryFormatter<
-  ParagraphAlertSingle[],
-  AlertSingle[]
-> = (entities: ParagraphAlertSingle[]) => {
-  return entities.map((entity) => ({
+export const formatter: QueryFormatter<ParagraphAlertSingle, AlertSingle> = (
+  entity: ParagraphAlertSingle
+) => {
+  return {
+    type: entity.type as AlertSingle['type'],
```

Comment:
It's an interesting question, I agree. Ultimately, yes, this (or something that serves the same purpose) is necessary, because of the way things are structured requiring the `type` property on all paragraphs (so that they can be mapped to their appropriate templates). When we define the TypeScript type, we require the property `type`, but we make it explicit what it can be:

```
export type AlertSingle = PublishedParagraph & {
  type: 'paragraph--alert_single'
  alertSelection: 'R' | 'NR'
  blockReference: AlertBlock
  nonReusableRef: AlertNonReusable
}
```

So, notably, it's _not_ a variable string. It must only be `paragraph--alert_single`. Meanwhile, `entity.type` _is_ a string. So if we pull out `as AlertSingle['type']`, then we get this error:

Reply:
To be very clear, it's this type of question (and this isn't the only one) that slowed my work significantly.